agent/lib_guzzle6.c

```diff
@@ -458,33 +408,142 @@ void nr_guzzle6_enable(TSRMLS_D) {
       "      return $promise;"
       "    };"
       "  }"
-      "}",
-      NULL, "newrelic/Guzzle6" TSRMLS_CC);
+      "}", guzzle_version, guzzle_version, guzzle_version);
+
+  char *guzzle_ver = nr_formatf("newrelic/Guzzle%d", guzzle_version);
+  _retval = zend_eval_string(eval, NULL, guzzle_ver TSRMLS_CC);
 
-  if (SUCCESS == retval) {
+  if (SUCCESS == _retval && guzzle_version == 6) {
     nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
-                              nr_guzzle_client_construct TSRMLS_CC);
-  } else {
+                              nr_guzzle6_client_construct TSRMLS_CC);
+  }else if (SUCCESS == _retval && guzzle_version == 7){
+    nr_php_wrap_user_function(NR_PSTR("GuzzleHttp\\Client::__construct"),
+                              nr_guzzle7_client_construct TSRMLS_CC);
+  }else {
     nrl_warning(NRL_FRAMEWORK,
                 "%s: error evaluating PHP code; not installing handler",
                 __func__);
   }
 }
 
-void nr_guzzle6_minit(TSRMLS_D) {
-  zend_class_entry ce;
+NR_PHP_WRAPPER_START(nr_guzzle_client_construct_helper){
```

Comment:
Maybe add a brief doc comment describing what this function does? I.e. it adds `newrelic\Guzzle\middleware` to the `GuzzleHttp\Client` handler stack.

Reply:
Good idea! I have added a brief description that describes what this function is doing [f3ac653](https://github.com/newrelic/newrelic-php-agent/pull/498/commits/f3ac6537477bfa2225380abeda4022fd29f4e1c4)